hydrator-common/src/main/java/io/cdap/plugin/common/LineageRecorder.java

```diff
@@ -36,11 +36,16 @@
 public class LineageRecorder {
 
   private final BatchContext context;
-  private final String dataset;
+  private Asset asset;
 
   public LineageRecorder(BatchContext context, String dataset) {
     this.context = context;
-    this.dataset = dataset;
+    this.asset = new Asset(dataset, null);
```

Comment:
do not pass null to a map, pass a `Collections.emptyMap()` or simply use the single parameter constructor

Reply:
Removed the map